src/utils/getSearchWith.ts

```diff
@@ -0,0 +1,21 @@
+import { Params } from '../types';
+
+export function getSearchWith(
+  params: Params,
+  search?: string | URLSearchParams,
+) {
+  const newParams = new URLSearchParams(search);
```

Comment:
You're creating a new instance of `URLSearchParams` from the `search` parameter. If `search` is undefined, this will create an empty `URLSearchParams` instance. Please make sure this is the intended behavior.